acceptance/tests/direct_puppet/static_catalog_env_control.rb

```diff
@@ -0,0 +1,348 @@
+test_name "Environment control of static catalogs" 
+require 'json'
+
+@testroot = master.tmpdir(File.basename(__FILE__, '/*'))
+### @testroot = "/etc/puppetlabs"
+
+@coderoot = "#{@testroot}/code"
+
+@master_opts = {
+  'main' => {
+    'environmentpath' => "#{@coderoot}/environments",
+  },
+}
+
+@agent_result_dirs = {}
+@catalog_files = {}
+
+agents.each do |agent|
+  @agent_result_dirs[agent.hostname] = agent.tmpdir('results')
+  @catalog_files[agent.hostname] = "#{on(agent, puppet('config', 'print', 'client_datadir')).stdout.chomp}/catalog/#{agent.hostname}.json"
+end
+
+@manifest = <<MANIFEST
+File {
+  ensure => directory,
+  mode => "0755",
+}
+
+file { 
+  '#{@testroot}':;
```

Comment:
This could be an array with `ensure` and `mode` specified, like
```
file {['#{@coderoot}',
        '#{@coderoot}/environments',
        ...]:
  ensure => directory,
  mode => '0755',
}
```

Then you could use the `File` declaration for all the files below and clean them up.